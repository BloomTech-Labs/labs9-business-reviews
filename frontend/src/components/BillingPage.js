import React, { Component } from 'react'
import styled from 'styled-components'
import SideBar from './SideBar';
import ProfileNav from './ProfileNav';

const StyledBilling = styled.div`
  width: 80%;
  display: flex;
  height: 600px;
  .billing{
    padding-left: 20px;
  }
  .payment-form{
    width: 280px;
    display: flex;
    flex-direction: column;

    border: 1px solid black;
    padding: 10px;
  }
  .payment-form input{
    width: 180px;
    margin: 10px;
    padding: 6px;
  }
  .input-div{
    display: flex;
    justify-content: center;
  }
  .checkbox{
    display: flex;
  }
  .checkbox p{
    margin-left: 5px;
  }
  button{
    padding: 30px;
    font-size: 20px;
    width: 300px;
    margin-top: 30px;
  }
`;

class BillingPage extends Component {
    constructor() {
        super();
    }

    render() {
      return(
        <div>
          <ProfileNav />
          <StyledBilling>
            <SideBar />
            <div className="billing">
              <h1>Billing</h1>
              <h4>Payment Info</h4>
              <form className="payment-form">
                <div className="input-div">
                  <p>CC#</p>
                  <input placeholder="####-####-####-####" />
                </div>
                <div className="input-div">
                  <p>EXP</p>
                  <input placeholder="MM/YY" />
                </div>
                <div className="input-div">
                  <p>CVV</p>
                  <input placeholder="###" />
                </div>
              </form>
              <form>
                <div className="checkbox">
                  <input type="checkbox" />
                  <p>1 Year Subscription - $9.99</p>
                </div>
                <div className="checkbox">
                  <input type="checkbox" />
                  <p>1 Month Subscription - $0.99</p>
                </div>
              </form>
              <button>Buy Now</button>
            </div>
          </StyledBilling>
        </div>
      )
    }
}

export default BillingPage;