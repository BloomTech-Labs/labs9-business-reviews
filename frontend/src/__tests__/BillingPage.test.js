import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import React from 'react';
import BillingPage from '../components/BillingPage';

describe('<BillingPage />', () => {
  it('renders', () => {
    shallow(<BillingPage />);
  });
  it('matches snapshot', () => {
    const wrapper = shallow(<BillingPage />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  it('should contain NavBar', () => {
    let app = shallow(<BillingPage />);
    expect(app.find("NavBar").length).toBe(1);
  });
  it('should contain BillingForm', () => {
    let app = shallow(<BillingPage />);
    expect(app.find("BillingForm").length).toBe(1);
  });
  it('should contain div with classname "billing"', () => {
    let app = shallow(<BillingPage/>);
    expect(app.find("div").length).toBe(2);
    expect(app.find(".billing").length).toBe(1);
  });
  it('should contain one button', () => {
    let app = shallow(<BillingPage/>);
    expect(app.find('button').length).toBe(1);
  });
});
