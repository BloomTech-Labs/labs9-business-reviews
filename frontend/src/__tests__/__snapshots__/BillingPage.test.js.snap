// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BillingPage /> matches snapshot 1`] = `
<div>
  <NavBar />
  <styled.div>
    <div
      className="billing"
    >
      <BillingForm>
        <button>
          Buy Now
        </button>
      </BillingForm>
    </div>
  </styled.div>
</div>
`;
